sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerSchema.java

```diff
@@ -55,7 +55,7 @@ private Builder addColumn(String table, Column column) {
     }
 
     public Builder addKeyPart(String table, String column, boolean desc) {
-      keyParts.put(table, KeyPart.create(column.toLowerCase(), desc));
+      keyParts.put(table.toLowerCase(), KeyPart.create(column.toLowerCase(), desc));
```

Comment:
This is a lot of toLowerCase's throughout the PR, seems very likely that someone will add new code that forgets to add toLowerCase and it will break. Is there anything else we can do? Maybe introduce a new type for case-insensitive strings and use it for table and column names? Not sure if that's better than the current state in terms of complexity though...